### PR TITLE
OPAL 639 Requeue API and Client

### PR DESCRIPTION
**Pull request by AM-USAF**

### CODE REVIEW

File: tests/test_api_endpoints.py [hunk 1]
```diff
@@ -76,13 +81,49 @@ def test_v1_queue_describe():
...
+        assert response.status_code == 200
+
 def test_v1_queue_lookup_item():
     """Tests the lookup_item endpoint.
     """
     queue.put(default_items)
 
     get = queue.get(1)
-
+
```
Comment (BE-USAF) [inline tests/test_api_endpoints.py:126]: whitespace

File: task_queue/in_memory_queue.py [hunk 2]
```diff
@@ -145,6 +145,32 @@ def fail(self, queue_item_id):
             self.memory_queue.fail,
             queue_item_id)
 
+    def wait(self):
```
Comment (BE-USAF) [inline task_queue/in_memory_queue.py:148]: While I like this functionality, I have two concerns with it.

1) If it is only for testing, maybe we should just implement it in the test section somehow.
2) I would be concerned with adding a new QueueBase type function without extending that to the other QueueBase implementations. I can see that being a source for potential bugs.

**Review by BE-USAF — COMMENTED** [review]
Still need to review the tests. Overall it looks great so far. I would strongly consider moving the move() logic into the clean_queue() testing logic

**Review by CK-USAF — DISMISSED** [review]

**Review by MM-USAF — APPROVED** [review]
Looks great, all pytest are passing

**Review by CK-USAF — APPROVED** [review]

**Review by MM-USAF — APPROVED** [review]
approved

**Review by CK-USAF — APPROVED** [review]

**Pull request merged**